### PR TITLE
OCPQE-11259  add build version for upgrade tests as image target value

### PR DESCRIPTION
**Pull request by dis016**
[OCPQE-11259](https://issues.redhat.com//browse/OCPQE-11259) Add build version for upgrade target.

### CODE REVIEW

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh [hunk 1]
```diff
@@ -12,3 +12,10 @@ if [[ -n "${RELEASE_IMAGE_LATEST}" ]]; then
     LATEST_VERSION="$(oc adm release info "${RELEASE_IMAGE_LATEST}" --output=jsonpath="{.metadata.version}")"
     echo "${LATEST_VERSION}" > "${BUILD_INFO_FILE}"
 fi
+if [[ -n "${RELEASE_IMAGE_TARGET}" ]]; then 
+    echo "${RELEASE_IMAGE_TARGET}"
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:16]: Can we remove this line? The line following it already shows the info

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh [hunk 2]
```diff
@@ -12,3 +12,10 @@ if [[ -n "${RELEASE_IMAGE_LATEST}" ]]; then
...
+    echo "Release info for: ${RELEASE_IMAGE_TARGET}"
+    oc adm release info "${RELEASE_IMAGE_TARGET}" || true
+    TARGET_VERSION="$(oc adm release info "${RELEASE_IMAGE_TARGET}" --output=jsonpath="{.metadata.version}")"
+    echo "${TARGET_VERSION}" > "${BUILD_INFO_FILE}"
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:20]: Would this override the `${BUILD_INFO_FILE}` and wipe out the `LATEST_VERSION` information?

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh (hunk 1 above)
Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:16]: @jhou1 added this line for debug information and while testing.

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh (hunk 2 above)
Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:20]: BUILD_INFO_FILE file will get overwritten with target version build number.

**Comment by dis016** [comment]
target build version is captured in artefacts
Screen Shot 2022-09-22 at 10 04 03 PM
image
 /cc @jhou1 @liangxia @pruan-rht  @JianLi-RH @chengzhang1016

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh (hunk 2 above)
Comment (liangxia) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:20]: We should use `>>` there, as we want to save both the from/to versions.

Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:20]: @liangxia ok, let me update it and send new commit

**Comment by jhou1** [comment]
OK, that looks right now.

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh (hunk 2 above)
Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh]: fixed in new commit

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh (hunk 1 above)
Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh]: fixed in new commit.

**Comment by dis016** [comment]
image
/cc: @jhou1 @liangxia @chengzhang1016

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh [hunk 3]
```diff
@@ -9,6 +9,12 @@ BUILD_INFO_FILE="${ARTIFACT_DIR}/ocp-build-info"
 if [[ -n "${RELEASE_IMAGE_LATEST}" ]]; then
     echo "Release info for: ${RELEASE_IMAGE_LATEST}"
     oc adm release info "${RELEASE_IMAGE_LATEST}" || true
-    LATEST_VERSION="$(oc adm release info "${RELEASE_IMAGE_LATEST}" --output=jsonpath="{.metadata.version}")"
-    echo "${LATEST_VERSION}" > "${BUILD_INFO_FILE}"
+    INITIAL_VERSION="$(oc adm release info "${RELEASE_IMAGE_LATEST}" --output=jsonpath="{.metadata.version}")"
+    echo "INITIAL_VERSION : ${INITIAL_VERSION}" >> "${BUILD_INFO_FILE}"
```
Comment (liangxia) [inline ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:13]: IMO, we should not add  the string `INITIAL_VERSION` in the file, it's hard for multi-version upgrade, e.g, we are required to add 4.3 -> 4.4 -> ... -> 4.12 upgrade

Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh]: fixed in new commit

File: ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh [hunk 4]
```diff
@@ -9,6 +9,12 @@ BUILD_INFO_FILE="${ARTIFACT_DIR}/ocp-build-info"
...
     oc adm release info "${RELEASE_IMAGE_LATEST}" || true
-    LATEST_VERSION="$(oc adm release info "${RELEASE_IMAGE_LATEST}" --output=jsonpath="{.metadata.version}")"
-    echo "${LATEST_VERSION}" > "${BUILD_INFO_FILE}"
+    INITIAL_VERSION="$(oc adm release info "${RELEASE_IMAGE_LATEST}" --output=jsonpath="{.metadata.version}")"
+    echo "${INITIAL_VERSION}" >> "${BUILD_INFO_FILE}"
 fi
+if [[ -n "${RELEASE_IMAGE_TARGET}" ]]; then 
+    echo "Release info for: ${RELEASE_IMAGE_TARGET}"
+    oc adm release info "${RELEASE_IMAGE_TARGET}" || true
+    UPGRADED_VERSION="$(oc adm release info "${RELEASE_IMAGE_TARGET}" --output=jsonpath="{.metadata.version}")"
```
Comment (liangxia) [inline ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh:18]: Prefer to use var name `TARGET_VERSION` here.
Consistent naming can help reading and debugging.

Comment (dis016) [reply ci-operator/step-registry/cucushift/upgrade/helper-service/build-info/cucushift-upgrade-helper-service-build-info-commands.sh]: @liangxia Thanks, fixed it.

**Pull request merged**